src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -102,22 +106,190 @@ export default class NativeAdapter implements IMediaSourceAdapter {
-    this._videoElement.src = "";
+    this._eventManager.destroy();
+    this._loadPromise = null;
+  }
+
+  /**
+   * Get the parsed tracks
+   * @function _getParsedTracks
+   * @returns {Array<Track>} - The parsed tracks
+   * @private
+   */
+  _getParsedTracks(): Array<Track> {
+    let videoTracks = this._getParsedVideoTracks();
+    let audioTracks = this._getParsedAudioTracks();
+    let textTracks = this._getParsedTextTracks();
+    return videoTracks.concat(audioTracks).concat(textTracks);
+  }
+
+  /**
+   * Get the parsed video tracks
+   * @function _getParsedVideoTracks
+   * @returns {Array<Track>} - The parsed video tracks
+   * @private
+   */
+  _getParsedVideoTracks(): Array<Track> {
+    let videoTracks = this._videoElement.videoTracks;
+    let parsedTracks = [];
+    if (videoTracks) {
+      for (let i = 0; i < videoTracks.length; i++) {
+        let settings = {
+          id: videoTracks[i].id,
+          active: videoTracks[i].selected,
+          label: videoTracks[i].label,
+          language: videoTracks[i].language,
+          index: i
+        };
+        parsedTracks.push(new VideoTrack(settings));
+      }
+    }
+    return parsedTracks;
+  }
+
+  /**
+   * Get the parsed audio tracks
+   * @function _getParsedAudioTracks
+   * @returns {Array<Track>} - The parsed audio tracks
+   * @private
+   */
+  _getParsedAudioTracks(): Array<Track> {
+    let audioTracks = this._videoElement.audioTracks;
+    let parsedTracks = [];
+    if (audioTracks) {
+      for (let i = 0; i < audioTracks.length; i++) {
+        let settings = {
+          id: audioTracks[i].id,
+          active: audioTracks[i].enabled,
+          label: audioTracks[i].label,
+          language: audioTracks[i].language,
+          index: i
+        };
+        parsedTracks.push(new AudioTrack(settings));
+      }
+    }
+    return parsedTracks;
+  }
+
+  /**
+   * Get the parsed text tracks
+   * @function _getParsedTextTracks
+   * @returns {Array<Track>} - The parsed text tracks
+   * @private
+   */
+  _getParsedTextTracks(): Array<Track> {
+    let textTracks = this._videoElement.textTracks;
+    let parsedTracks = [];
+    if (textTracks) {
+      for (let i = 0; i < textTracks.length; i++) {
+        let settings = {
+          kind: textTracks[i].kind,
+          active: textTracks[i].mode === 'showing',
+          label: textTracks[i].label,
+          language: textTracks[i].language,
+          index: i
+        };
+        parsedTracks.push(new TextTrack(settings));
+      }
+    }
+    return parsedTracks;
+  }
+
+  /**
+   * Select a video track
+   * @function selectVideoTrack
+   * @param {VideoTrack} videoTrack - the track to select
+   * @returns {boolean} - success
+   * @public
+   */
+  selectVideoTrack(videoTrack: VideoTrack): boolean {
+    let videoTracks = this._videoElement.videoTracks;
+    if ((videoTrack instanceof VideoTrack) && videoTracks && videoTracks[videoTrack.index]) {
```

Comment:
@yairans please verify if it is better to first disable all tracks and only then enable the new one, same for all track types.
That way you can create a method to disable tracks which just goes through all tracks and disable, and then just access the chosen track by index and set it directly as enabled.